frontend/src/utils/subtitles.ts

```diff
@@ -0,0 +1,162 @@
+  return hours * 3600 + minutes * 60 + seconds;
+}
+
+/**
+* Parses a VTT (WebVTT) file from a given URL
+* Extracts dialogue lines with start and end times, and text content.
+* 
+* Converts specific tags to styled <span> tags, 
+* and sanitizes the content by removing harmful tags.
+*/
+export async function parseVttFile(src: string) {
+    return fetch(src)
+      .then(response => response.text())
+      .then((vttText) => {
+        try {
+          const subtitles: ParsedSubtitleTrack = [];
+          const vttLines = vttText.split('\n');
+
+          let i = 0;
+          while (i < vttLines.length) {
+            // Skip empty lines
+            if (vttLines[i].trim() === '') {
+              i++;
+              continue;
+            }
+
+            if (vttLines[i].includes('-->')) {
+              const [start, end] = vttLines[i].split(' --> ');
+              let text = '';
+              i++;
+              while (i < vttLines.length && !vttLines[i].includes('-->')) {
+                text += vttLines[i] + '\n';
+                i++;
+              }
+
+              // Replace tags with html elements & remove harmful tags to sanitize
+              const sanitizedText = text
+                .replace(/<i>/g, '<span style="font-style: italic;">')
+                .replace(/<\/i>/g, '</span>')
+                .replace(/<b>/g, '<span style="font-weight: bold;">')
+                .replace(/<\/b>/g, '</span>')
+                .replace(/<u>/g, '<span style="text-decoration: underline;">')
+                .replace(/<\/u>/g, '</span>')
+                .replace(/<em>/g, '<span style="font-style: italic;">')
+                .replace(/<\/em>/g, '</span>')
+                .replace(/<strong>/g, '<span style="font-weight: bold;">')
+                .replace(/<\/strong>/g, '</span>')
+                .replace(/<mark>/g, '<span style="background-color: yellow;">')
+                .replace(/<\/mark>/g, '</span>')
+                .replace(/<script.*?>.*?<\/script>/g, '')
+                .replace(/<iframe.*?>.*?<\/iframe>/g, '')
+                .replace(/<object.*?>.*?<\/object>/g, '')
+                .replace(/<embed.*?>.*?<\/embed>/g, '')
+                .replace(/<style.*?>.*?<\/style>/g, '');
+
+              subtitles.push({
+                start: parseTime(start),
+                end: parseTime(end),
+                text: sanitizedText.trim()
+              });
+            } else {
+              i++;
+            }
+          }
+
+          return subtitles;
+        } catch (err) {
+          console.error("error parsing VTT subtitles", err);
+        }
+      });
+  }
+
+/**
+* Parses an ASS/SSA (SubStation Alpha) file from a given URL.
+* Extracts dialogue lines with start and end times, and text content.
+* 
+* Converts specific tags to styled <span> tags, 
+* and sanitizes the content by removing harmful tags.
+*/
+export async function parseSsaFile(src: string) {
```

Comment:
Same as above. This is so similar to above function, really you don't think the common parts can be refactored out and shared between both? See for instance the ``api.ts`` file from composables, where there's a lot of extracted parts out there for some insights!

Reply:
I refactored the SSA/ASS parsing. I was more focused on writing out VTT first since it's the more popular format and much simpler, so most of the code was reused and not complete. Now it should take the specified format into account and add multi-line dialogue together.